src/migrate/index.js

```diff
@@ -108,9 +115,49 @@ export default class Migrator {
     });
   }
 
+  _createMigrationLockTable(tableName) {
+    return this.knex.schema.createTable(tableName, function(t) {
+      t.boolean('is_locked');
+    });
+  }
+
+  _getLockTableName(tableName) {
+    return tableName + '_lock';
+  }
+
+  _isLocked() {
+    var tableName = this._getLockTableName(this.config.tableName);
+    return this.knex(tableName)
+      .count('is_locked as count')
+      .then(function(data) {
+        if (data[0].count > 0) {
+          return true;
+        }
+        return false;
+      });
+  }
+
+  _lockMigrations() {
+    var tableName = this._getLockTableName(this.config.tableName);
+    return this.knex(tableName)
+      .insert({ is_locked: true });
+  }
+
+  _unlockMigrations() {
+    var tableName = this._getLockTableName(this.config.tableName);
+    return this.knex(tableName).del();
+  }
+
   // Run a batch of current migrations, in sequence.
   _runBatch(migrations, direction) {
-    return Promise.all(_.map(migrations, this._validateMigrationStructure, this))
+    return this._isLocked()
+      .then((isLocked) => {
+        if (isLocked) {
+          throw new Error('migrations failed: migration in progress');
```

Comment:
You can here create own Error e.g. LockingError() type for this, in that way you can have separate error handler in check examples from here: http://bluebirdjs.com/docs/api/catch.html.